common/src/main/java/com/wynntils/screens/OverlayManagementScreen.java

```diff
@@ -0,0 +1,651 @@
+        if (testMode) return super.mouseClicked(mouseX, mouseY, button);
+
+        // Order:
+        //  - Corners
+        //  - Edges
+        //  - OverlayArea
+
+        // reset
+        resetSelection();
+
+        Overlay overlay = selectedOverlay;
+
+        Vec2 mousePos = new Vec2((float) mouseX, (float) mouseY);
+
+        for (Map.Entry<Corner, Vec2> corner : overlay.getCornersMap().entrySet()) {
+            float distance = corner.getValue().distanceToSqr(mousePos);
+            if (distance < MAX_CLICK_DISTANCE) {
+                selectedCorner = corner.getKey();
+                selectionMode = SelectionMode.Corner;
+
+                return false;
+            }
+        }
+
+        for (Edge value : Edge.values()) {
+            float minX, maxX, minY, maxY;
+
+            switch (value) {
+                case Top -> {
+                    minX = overlay.getRenderX();
+                    maxX = overlay.getRenderX() + overlay.getWidth();
+                    minY = overlay.getRenderY() - MAX_CLICK_DISTANCE / 2f;
+                    maxY = overlay.getRenderY() + MAX_CLICK_DISTANCE / 2f;
+                }
+                case Left -> {
+                    minX = overlay.getRenderX() - MAX_CLICK_DISTANCE / 2f;
+                    maxX = overlay.getRenderX() + MAX_CLICK_DISTANCE / 2f;
+                    minY = overlay.getRenderY();
+                    maxY = overlay.getRenderY() + overlay.getHeight();
+                }
+                case Right -> {
+                    minX = overlay.getRenderX() + overlay.getWidth() - MAX_CLICK_DISTANCE / 2f;
+                    maxX = overlay.getRenderX() + overlay.getWidth() + MAX_CLICK_DISTANCE / 2f;
+                    minY = overlay.getRenderY();
+                    maxY = overlay.getRenderY() + overlay.getHeight();
+                }
+                case Bottom -> {
+                    minX = overlay.getRenderX();
+                    maxX = overlay.getRenderX() + overlay.getWidth();
+                    minY = overlay.getRenderY() + overlay.getHeight() - MAX_CLICK_DISTANCE / 2f;
+                    maxY = overlay.getRenderY() + overlay.getHeight() + MAX_CLICK_DISTANCE / 2f;
+                }
+                default -> {
+                    // should not happen
+                    continue;
+                }
+            }
+
+            if (mouseX >= minX && mouseX <= maxX && mouseY >= minY && mouseY <= maxY) {
+                selectedEdge = value;
+                selectionMode = SelectionMode.Edge;
+
+                return false;
+            }
+        }
+
+        if (isMouseHoveringOverlay(selectedOverlay, mouseX, mouseY)) {
+            selectionMode = SelectionMode.Area;
+
+            return false;
+        }
+
+        return super.mouseClicked(mouseX, mouseY, button);
+    }
+
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        if (testMode) return false;
```

Comment:
So for pressed in testMode we pass it on to super, but not for release? Is this intentional? Can't we get "stuck" on some background UI element?

Reply:
Explained why mousePressed is important above, this has no use case when in test mode.